File: src/main/view/systemIntegrations/src/app/matrix-cell/MatrixCell.ts
import {Coordinates} from './Coordinates';

export class MatrixCell {
  private _coordinates: Coordinates;
  private _displayedInfo: string;

  constructor(x: number, y: number, displayedInfo: string) {
    this._coordinates = new Coordinates(x, y);
    this._displayedInfo = displayedInfo;
  }

  get coordinates(): Coordinates {
    return this._coordinates;
  }

  set coordinates(value: Coordinates) {
    this._coordinates = value;
  }

  get displayedInfo(): string {
    return this._displayedInfo;
  }

  set displayedInfo(value: string) {
    this._displayedInfo = value;
  }
}

File: src/main/java/ru/ilka/systemintegration/entity/MatrixCell.java
package ru.ilka.systemintegration.entity;

public class MatrixCell {
    private Coordinates coordinates;
    private String displayedInfo;

    public MatrixCell() {
        displayedInfo = "   ";
    }

    public MatrixCell(int x, int y, String displayedInfo) {
        this.coordinates = new Coordinates(x, y);
        this.displayedInfo = displayedInfo;
    }

    public MatrixCell(Coordinates coordinates, String displayedInfo) {
        this.coordinates = coordinates;
        this.displayedInfo = displayedInfo;
    }

    public Coordinates getCoordinates() {
        return coordinates;
    }

    public void setCoordinates(Coordinates coordinates) {
        this.coordinates = coordinates;
    }

    public String getDisplayedInfo() {
        return displayedInfo;
    }

    public void setDisplayedInfo(String displayedInfo) {
        this.displayedInfo = displayedInfo;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        MatrixCell that = (MatrixCell) o;

        if (!coordinates.equals(that.coordinates)) return false;
        return displayedInfo.equals(that.displayedInfo);
    }

    @Override
    public int hashCode() {
        int result = coordinates.hashCode();
        result = 31 * result + displayedInfo.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "MatrixCell{" +
                "coordinates=" + coordinates +
                ", displayedInfo='" + displayedInfo + '\'' +
                '}';
    }
}
